vars/infra.groovy

```diff
@@ -142,6 +142,8 @@ Object checkoutSCM(String repo = null) {
   // Enable long paths to avoid problems with tests on Windows agents
   if (!isUnix()) {
     bat 'git config --global core.longpaths true'
+    // TODO: Remove when https://github.com/jenkins-infra/helpdesk/issues/3865 is resolved
+    bat 'git config --global core.eol lf'
```

Comment:
the spotless issue is that it ends up just guessing that the line ending is native becasue there is no configuration otherwise.

The alternative (which uses windows line ends on windows, so is more like what windows would do) is to set `git config --global core.autocrlf true` 

this should cause the files to have line ends changed on windows, and will be picked up by spotless to go "oh it is converting line ends, and I am on windows" so spotless does not end up guessing.

Reply:
Thanks very, very much!  Your guidance was exactly what I needed.  I confirmed with several tests that `core.autocrlf` resolves the spotless formatting issue and allows builds to be portable between ci.jenkins.io Windows containers and Windows virtual machines